### ISSUE TITLE
Save bitmap to clipboard

### ISSUE THREAD

**Issue by techtonik** [asweigart/pyperclip#48]
It would be nice if there was API that can save captured bitmaps to clipboard. Something that accepts PyCBitmap object from pywin32:

    SetClipboardData(CF_BITMAP, dataBitMap)

**Comment by asweigart** [comment]
Good idea. I'd like to keep Pyperclip platform-independent. But if it's not that difficult to implement this on Windows, then having this be a Windows-only feature for now would be good.

**Comment by mhils** [comment]
Implementing this is quite easy - one just needs to expose a `format` parameter which defaults to `CF_UNICODETEXT` and then replace all occurences of `CF_UNICODETEXT` with that.
The harder problem is that different function signatures per OS are really ugly. One could add the `format` parameter to every clipboard and raise RuntimeErrors if the clipboard is text only add a dedicated `copy_bitmap` method. The latter is difficult again because there isn't really a finite list of clipboard data types (https://msdn.microsoft.com/en-us/library/windows/desktop/ms649013(v=vs.85).aspx)

**Comment by techtonik** [comment]
Well, ability of clipboard formats is not a problem - as long as there is room for extension people will add what they need when they need it. The current problem is to understand where to add this and how to debug possible "unable to convert format XXX to YYY" errors. At least information about XXX and YYY should be easily accessible.

My fullstop was that I don't know how to convert PyCBitmap wrapper from pywin32 back into the system level object to call SetClipboardData WinAPI through ctypes.

**Comment by mhils** [comment]
The current implementation of `copy` should handle memory allocation just fine for all cases. If you look at the way it works, there's no actual conversion (just copying which would need to be fixed to work with non-unicode lengths, probably by just converting unicode to bytes in the Python level)

**Comment by jaraco** [comment]
See the [jaraco.clipboard](https://pypi.io/project/jaraco.clipboard) for an implementation that handles clipboard images on Windows.

**Comment by user-e** [comment]
This would be a useful feature - having a way to _[copy/get an image in the clipboard with Python](https://stackoverflow.com/questions/24474536/how-to-copy-get-an-image-in-the-clipboard-with-python-i-accept-tkinter-for-text)_, using this simple cross-platform clipboard module.
However, the _jaraco.clipboard_ module described above appears to do the job, with HTML and image formats supported on MacOS and Windows (but not Linux?).

**Comment by stuaxo** [comment]
Under Linux would format end up being a mimetype ?

See -
https://askubuntu.com/questions/759651/how-to-copy-an-image-to-the-clipboard-from-a-file-using-command-line
